lib/src/util/posix/scoped_bio.cc

```diff
@@ -6,8 +6,11 @@ using namespace leatherman::util;
 
 namespace facter { namespace util { namespace posix {
 
-    scoped_bio::scoped_bio(BIO_METHOD* method) :
-        scoped_resource(BIO_new(method), free)
+    // Remove const-ness before calling BIO_new. This is "unsafe",
+    // but in isolation here will not cause issues. Allows the code to work
+    // with both OpenSSL 1.0 and 1.1.
+    scoped_bio::scoped_bio(const BIO_METHOD* method) :
+        scoped_resource(BIO_new(const_cast<BIO_METHOD*>(method)), free)
```

Comment:
This is messy, but I feel more maintainable than ifdefing based on OpenSSL version (because it doesn't branch code).